GVFS/GVFS.Virtualization/FileSystemCallbacks.cs

```diff
@@ -89,6 +89,9 @@ public FileSystemCallbacks(GVFSContext context, GVFSGitObjects gitObjects, RepoM
                 throw new InvalidRepoException(error);
             }
 
+            this.modifiedPaths.Compress(this.context.Tracer);

```

Comment:
This is a lot of work to be doing in a constructor, and it can have surprising consequences. Is there anywhere that we can do this work after this point?

Reply:
We can move it to the `TryStart` method.